common/interpolation/include/interpolation/spline_interpolation.hpp

```diff
@@ -40,9 +45,86 @@ struct MultiSplineCoef
   std::vector<double> d;
 };
 
+// static spline interpolation functions
 std::vector<double> slerp(
   const std::vector<double> & base_keys, const std::vector<double> & base_values,
   const std::vector<double> & query_keys);
+
+// std::vector<double> slerpDiff(
+//   const std::vector<double> & base_keys, const std::vector<double> & base_values,
+//   const std::vector<double> & query_keys);
+
+// TODO(murooka) use template
+// template <typename T>
+// std::vector<double> slerpYawFromPoints(const std::vector<T> & points);
+std::vector<double> slerpYawFromPoints(const std::vector<geometry_msgs::msg::Point> & points);
 }  // namespace interpolation
 
+// non-static 1-dimensional spline interpolation
+//
+// Usage:
+// ```
+// SplineInterpolation1d spline;
+// spline.calcSplineCoefficients(base_keys, base_values);  // memorize pre-interpolation result
+// internally
+// const auto interpolation_result1 = spline.getSplineInterpolatedValues(base_keys,
+//   query_keys1);
+// const auto interpolation_result2 = spline.getSplineInterpolatedValues(base_keys,
+//   query_keys2);
+// ```
+class SplineInterpolation1d
+{
+public:
+  SplineInterpolation1d() = default;
+
+  void calcSplineCoefficients(
+    const std::vector<double> & base_keys, const std::vector<double> & base_values);
+
+  std::vector<double> getSplineInterpolatedValues(const std::vector<double> & query_keys) const;
+
+private:
+  std::vector<double> base_keys_;
+  interpolation::MultiSplineCoef multi_spline_coef_;
+};
+
+// non-static points spline interpolation
+// NOTE: We can calculate yaw from the x and y by interpolation derivatives.
+//
+// Usage:
+// ```
+// SplineInterpolationPoint2d spline;
+// spline.calcSplineCoefficients(base_keys, base_values);  // memorize pre-interpolation result
+// internally
+// const auto interpolation_result1 = spline.getSplineInterpolatedPoints(base_keys,
+//   query_keys1);
+// const auto interpolation_result2 = spline.getSplineInterpolatedPoints(base_keys,
+//   query_keys2);
+// const auto yaw_interpolation_result = spline.getSplineInterpolatedYaws(base_keys,
+//   query_keys1);
+// ```
+class SplineInterpolationPoint2d
```

Comment:
2-dimensional point (z is ignored) spline interpolation
Yaw can be calculated by spline interpolation derivative.